src/pty.c

```diff
@@ -0,0 +1,118 @@
+#define _BSD_SOURCE
+#define _DEFAULT_SOURCE
+#include <errno.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <signal.h>
+#include <pty.h>
+#include <re.h>
+#include "test.h"
+
+
+#define DEBUG_MODULE "test_pty"
+#define DEBUG_LEVEL 5
+#include <re_dbg.h>
+
+
+struct pty_test {
+	int pty;
+	pid_t pid;
+	int err;
+};
+
+
+static void abort_test(struct pty_test *test, int err)
+{
+	test->err = err;
+	re_cancel();
+}
+
+
+static void pty_hdlr(int flags, void *arg)
+{
+	struct pty_test *test = arg;
+	int err;
```

Comment:
nit; `int err` is uninitialized.

Reply:
I think it's always set before it's evaluated. Do you want me to initialise it with `0`?